### PR TITLE
[Fix] : replace point locks with table lock in some cases (#15578)

### PR DESCRIPTION
**Pull request by jensenojs**
Previous pr's used the wrong stats, now it's standardized to do the judgement in the last step of building the logic plan.

## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue # https://github.com/matrixorigin/MO-Cloud/issues/3006

## What this PR does / why we need it:

sql like `insert into A select * from B`, will try to lock the value in `B`. When the number of these locks is too much, it will lead to bugs reflected in the issue. In the context of range locks being temporarily unavailable, use table lock to replace point lock

### CODE REVIEW

File: pkg/sql/plan/build_util.go [hunk 1]
```diff
@@ -41,6 +42,31 @@ import (
...
+	lockconfig := lockService.GetConfig()
+
+	for _, n := range builder.qry.Nodes {
+		if n.NodeType != plan.Node_LOCK_OP {
+			continue
+		}
+		if !n.LockTargets[0].LockTable {
+			reCheckIfNeed := n.Stats.Outcnt > float64(lockconfig.MaxLockRowCount)
+			if reCheckIfNeed {
+				logutil.Warnf("Row lock upgraded to table lock for SQL : %s", builder.compCtx.GetRootSql())
```
Comment (ouyuanning) [inline pkg/sql/plan/build_util.go:62]: why use warn not info?

File: pkg/sql/plan/query_builder.go [hunk 2]
```diff
@@ -2854,6 +2854,10 @@ func (builder *QueryBuilder) buildSelect(stmt *tree.Select, ctx *BindContext, is
 	if isRoot {
 		builder.qry.Headings = append(builder.qry.Headings, ctx.headings...)
 	}
+	
+	if builder.isForUpdate {
```
Comment (ouyuanning) [inline pkg/sql/plan/query_builder.go:2858]: 建议放在优化器那部分代码的最后。作为一种优化规则

File: pkg/sql/plan/build_util.go (hunk 1 above)
Comment (jensenojs) [reply pkg/sql/plan/build_util.go:62]: ok

**Comment by jensenojs** [comment]
# select for update

this pr 
Pasted Graphic 7

**Comment by jensenojs** [comment]
daiily测试link : https://github.com/matrixorigin/mo-nightly-regression/actions/runs/8874874327/job/24363452244

**Review by zhangxu19830126 — APPROVED** [review]

**Review by reusee — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Pull request merged**